src/github.com/t3chguy/riot-static/riot-static.go

```diff
@@ -1,227 +0,0 @@
-// Copyright 2017 Michael Telatynski <7t3chguy@gmail.com>
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package main
-
-import (
-	"fmt"
-	"github.com/gin-gonic/gin"
-	"github.com/t3chguy/riot-static/mxclient"
-	"github.com/t3chguy/riot-static/sanitizer"
-	"github.com/t3chguy/riot-static/templates"
-	"github.com/t3chguy/riot-static/utils"
-	"net/http"
-	"os"
-	"time"
-)
-
-// TODO Cache memberList+serverList until it changes
-
-const PublicRoomsPageSize = 20
-const RoomTimelineSize = 30
-const RoomMembersPageSize = 20
-
-const NumWorkers uint32 = 32
-
-func main() {
-	client := mxclient.NewClient()
-	worldReadableRooms := client.NewWorldReadableRooms()
-
-	workers := NewWorkers(NumWorkers, client)
-	sanitizer := sanitizer.InitSanitizer()
-
-	router := gin.Default()
-	router.Static("/img", "./assets/img")
-
-	router.GET("/", func(c *gin.Context) {
-		page := utils.StrToIntDefault(c.DefaultQuery("page", "1"), 1)
-		templates.WritePageTemplate(c.Writer, &templates.RoomsPage{
-			Rooms:    worldReadableRooms.GetPage(page, PublicRoomsPageSize),
-			PageSize: PublicRoomsPageSize,
-			Page:     page,
-		})
-	})
-
-	roomRouter := router.Group("/room/:roomID/")
-	{
-		// Load room worker into request object so that we can do any clean up etc here
-		roomRouter.Use(func(c *gin.Context) {
-			roomID := c.Param("roomID")
-			worker := workers.GetWorkerForRoomID(roomID)
-
-			worker.Queue <- &RoomInitialSyncJob{roomID}
-			resp := (<-worker.Output).(*RoomInitialSyncResp)
-
-			if resp.err != nil {
-				c.String(http.StatusNotFound, "Room Not Found")
-				c.Abort()
-				return
-			}
-
-			c.Set("RoomWorker", worker)
-			c.Next()
-
-			//	c.HTML(http.StatusInternalServerError, "room_error.html", gin.H{
-			//		"Error": "Failed to load room.",
-			//		"Room":  room,
-			//	})
-		})
-
-		roomRouter.GET("/", func(c *gin.Context) {
-			worker := c.MustGet("RoomWorker").(Worker)
-			offset := utils.StrToIntDefault(c.DefaultQuery("offset", "0"), 0)
-			eventID := c.DefaultQuery("anchor", "")
-
-			worker.Queue <- Job(RoomEventsJob{
-				c.Param("roomID"),
-				eventID,
-				offset,
-				RoomTimelineSize,
-			})
-
-			jobResult := (<-worker.Output).(RoomEventsResp)
```

Comment:
See earlier comment on `select`. We probably want to give up waiting for a response after a while.